servers/cu/src/routes/middleware/withProcessRestriction.js

```diff
@@ -1,15 +1,15 @@
 const withProcessRestrictionFrom = ({ extractor }) => (handler) => (req, res, next) => {
   const {
     domain: {
-      RESTRICT_PROCESSES
-      // TODO: add RESTRICT_TO_PROCESSES and check below
-      // See https://github.com/permaweb/ao/pull/587
+      RESTRICT_PROCESSES,
+      ALLOW_PROCESSES
     }
   } = req
 
   const processId = extractor(req)
 
   if (RESTRICT_PROCESSES && RESTRICT_PROCESSES.includes(processId)) return res.status(403).send({ error: `Access denied for process ${processId}` })
+  if (ALLOW_PROCESSES && !ALLOW_PROCESSES.includes(processId)) return res.status(403).send({ error: `Access denied for process ${processId}` })
```

Comment:
Since this is a `whitelist`, I think this will need to also check `length`. Otherwise, I think an empty list will bar all processes.

Reply:
you are absolutely right